contentcuration/contentcuration/views/nodes.py

```diff
@@ -165,6 +180,153 @@ def get_nodes_by_ids_complete(request, ids):
     return Response(serializer.data)
 
 
+def get_channel_thumbnail(channel):
+    # Problems with json.loads, so use ast.literal_eval to get dict
+    if channel.get("thumbnail_encoding"):
+        thumbnail_data = ast.literal_eval(channel.get("thumbnail_encoding"))
+        if thumbnail_data.get("base64"):
+            return thumbnail_data["base64"]
+
+    if channel.get("thumbnail"):
+        return generate_storage_url(channel.get("thumbnail"))
+
+def get_thumbnail(node):
+    # Problems with json.loads, so use ast.literal_eval to get dict
+    if node.thumbnail_encoding:
+        thumbnail_data = ast.literal_eval(node.thumbnail_encoding)
+        if thumbnail_data.get("base64"):
+            return thumbnail_data["base64"]
+
+    thumbnail = node.files.filter(preset__thumbnail=True).first()
+    if thumbnail:
+        return generate_storage_url(str(thumbnail))
+
+    return "/".join([settings.STATIC_URL.rstrip("/"), "img", "{}_placeholder.png".format(node.kind_id)])
+
+
+DATE_TIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+@api_view(['GET'])
+def get_topic_details(request, contentnode_id):
+    """ Generates data for topic contents. Used for look-inside previews
+        Keyword arguments:
+            contentnode_id (str): id of topic node to get details from
+    """
+    # Get nodes and channel
+    node = ContentNode.objects.prefetch_related('children', 'files', 'tags')\
+                            .select_related('license', 'language')\
+                            .get(pk=contentnode_id)
+    descendants = node.get_descendants()
+    channel = node.get_channel()
+
+    # If channel is a sushi chef channel, use date created for faster query
+    # Otherwise, find the last time anything was updated in the channel
+    last_update = channel.main_tree.created if channel and channel.ricecooker_version else \
+                        descendants.filter(changed=True)\
+                                .aggregate(latest_update=Max('modified'))\
+                                .get('latest_update')
+
+    # See if the latest cached data is up to date since the last update to the channel
+    cached_data = cache.get("details_{}".format(node.node_id))
```

Comment:
Do we now need to support django caches in the backend? My main worry is that if requests are serviced in one and then in another, the cache might not be shared by default among instances.